File: angular/src/models/media-model.ts
import { GroupModel } from "./group-model";

export interface MediaModel {
  id: string;
  type: string;
  owner: string;
  description?: any;
  serialNumber: string;
  groupId: string;
  group: GroupModel;
  modelType: string;
  searchingWeight: number;
}

File: aspnet-core/src/SimonsVoss.SearchableFields.Web/Models/MediaModel.cs
using SimonsVoss.SearchableFields.Web.Core;

namespace SimonsVoss.SearchableFields.Web.Models;

public class MediaModel : SearchableModel<Guid>
{
    public string Type { get; set; }
    public string Owner { get; set; }
    public string Description { get; set; }
    public string SerialNumber { get; set; }

    public Guid? GroupId { get; set; }
    public GroupModel Group { get; set; }

    public override string ModelType => "Media";
}